### PR TITLE
Loader fix

### LINKED ISSUE #58331: [BUG] updating a Xen paravirtualized guest removes the kernel tag
**Description**

Xen Paravirtualized VMs have an `os/kernel` element like the following:

```
  <os>
    <type arch='x86_64'>linux</type>
    <kernel>/usr/lib/grub2/x86_64-xen/grub.xen</kernel>
  </os>
```

When running `virt.update` with `boot=None` which is the default, the kernel element is removed instead of being left untouched.

### PR DESCRIPTION
**Pull request by cbosdo**
### What does this PR do?

Refactor the way the virtual machine description XML is modified in `virt.udpate()`. This has two advantages:
  * it makes it easier to modify more values in the future
  * properly handles the boot parameters modifications (issue #58331)

### What issues does this PR fix or reference?
Fixes: #58331

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs
- [X] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by Akm0d** [comment]
re-run full all

**Review by Akm0d — DISMISSED** [review]

**Comment by krionbsd** [comment]
re-run all

**Comment by krionbsd** [comment]
re-run all

**Comment by cbosdo** [comment]
Rebased on master, fixed the black issues (how comes they weren't there yesterday?) and add the changelog file that I forgot to `git add`

**Comment by cbosdo** [comment]
rebased on master, removed the black changes due to new version rollback.

File: tests/unit/utils/test_data.py [hunk 1]
```diff
@@ -1498,3 +1489,59 @@ def test_ignore_missing_keys_recursive(self):
                 dict_one, dict_two, ignore_missing_keys=True
             ),
         )
+
+
+def test_get_value_simple_path():
```
Comment (s0undt3ch) [inline tests/unit/utils/test_data.py:1494]: Please place these tests under `tests/pytests`

File: tests/unit/utils/test_xmlutil.py [hunk 2]
```diff
@@ -171,3 +169,169 @@ def test_xml_case_f_full(self):
         xmldata = ET.fromstring(self.cases["f"]["xml"])
         defaultdict = xml.to_dict(xmldata, True)
         self.assertEqual(defaultdict, self.cases["f"]["full"])
+
+
+@pytest.fixture
```
Comment (s0undt3ch) [inline tests/unit/utils/test_xmlutil.py:174]: Please place these under `tests/pytests`

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by cbosdo** [comment]
@s0undt3ch rebased on master and moved the new tests to the `tests/pytest` folder

**Comment by cbosdo** [comment]
@s0undt3ch @Akm0d all tests green and the requested changes are in. Will hit the 'update branch' button

**Review by Akm0d — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Comment by cbosdo** [comment]
@dwoz since it's all green now, could you merge it?

**Pull request merged**